demo/dlb-init.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+enable_and_configure_vfs() {
+  devpath=$1
+  num_vfs_per_pf=$2
+  # enable sriov
+  sriov_numvfs_path="$devpath"/sriov_numvfs
+  if test -w "$sriov_numvfs_path"; then
+    if [ "$(cat "$sriov_numvfs_path")" -eq 0 ]; then
+     echo -n "$num_vfs_per_pf" > "$sriov_numvfs_path"
+      # conffigure vfs
+      for virtfnN in $(basename -a "$devpath"/virtfn*) ; do
+        #unbind vf
+        vf_pciid=$(basename "$(realpath "$devpath"/"$virtfnN")")
+        dlb_pci_driver_path=/sys/bus/pci/drivers/dlb2
+        echo -n "$vf_pciid" > $dlb_pci_driver_path/unbind
+
+        #configure vf
+        vf_resources_path="$devpath"/vf${virtfnN#virtfn}_resources
+       echo -n 2048 > "$vf_resources_path"/num_atomic_inflights
+       echo -n 2048 > "$vf_resources_path"/num_dir_credits
+       echo -n 8 > "$vf_resources_path"/num_dir_ports
+       echo -n 2048 > "$vf_resources_path"/num_hist_list_entries
+       echo -n 8192 > "$vf_resources_path"/num_ldb_credits
+       echo -n 4 > "$vf_resources_path"/num_ldb_ports
+       echo -n 32 > "$vf_resources_path"/num_ldb_queues
+       echo -n 32 > "$vf_resources_path"/num_sched_domains
+       echo -n 2 > "$vf_resources_path"/num_sn0_slots
+       echo -n 2 > "$vf_resources_path"/num_sn1_slots
+
+        #bind vf back to dlb2 driver
+        echo -n "$vf_pciid" > $dlb_pci_driver_path/bind
+      done
+     echo "$devpath" configured
+    else
+     echo "$devpath" already configured
+    fi
+  else
+   echo "$sriov_numvfs_path" "is not found. Check if dlb driver is configured"
```

Comment:
try to check error conditions so that error message is closer to the check, e.g.
```
if <error condition> ; then
  echo 'error: some error, skip processing'
else
   echo 'no error, can continue'
fi
```

Reply:
Sorry, but what <error condition> can be? Is there a way to get what error it is?

Reply:
Sorry, but I do not see the point of this review.! Please let me know if adding "error: " is not enough.!

Reply:
> Please let me know if adding "error: " is not enough.!

the feedback was not about this at all. to me the comment is very clear: the error condition is that the file is not writable, so better to echo the error right away and only continue if it's OK. now the error output comes after 30 lines

Reply:
@hj-johannes-lee
> Sorry, but I do not see the point of this review.!

The point of this review is to make your code more readable and understandable.
In this particular case it means moving error output closer to error condition. Having it 30 lines below is less readable and understandable.

Reply:
@bart0sh Ah, now I understand.! Thank you for your detailed explanation always.! :)